packages/approval-controller/src/ApprovalController.ts

```diff
@@ -90,8 +91,10 @@ export type ApprovalRequest<RequestData extends ApprovalRequestData> = {
 
   /**
    * The type of the approval request.
+   * Unfortunately, not all values will match the `ApprovalType` enum.
+   * TODO: Remove the `string` member by either narrowing the `type` property, or expanding the `ApprovalType` enum, to encompass all approval request types used in the clients.
    */
-  type: string;
+  type: ApprovalType | string;
```

Comment:
Hmm. This change now seems to have no effect; the type of this property is still `string`. This happens because `string` is a superset of `ApprovalType`.

Reply:
hmm.. good pointing this out. Also nice to have both to be explicit and helpful in conjunction with the todo comment

Reply:
Oh okay, that's fine then.